packages/app/src/elements/import-dialog.ts

```diff
@@ -40,14 +97,126 @@ export class ImportDialog extends Dialog<File, void> {
                 ></pl-select>
 
                 <div class="small padded" ?hidden=${this._formatSelect && this._formatSelect.value !== imp.CSV.value}>
-                    ${$l(
-                        "IMPORTANT: Before importing, please make sure that your CSV data " +
-                            "is structured according to {0}'s specific requirements!",
-                        process.env.PL_APP_NAME!
-                    )}
-                    <a href="#" @click=${this._downloadCSVSampleFile}> ${$l("Download Sample File")} </a>
+                    ${$l("Choose the correct column names and types for each column below. If you are having trouble,")}
+                    <a href="#" @click=${this._downloadCSVSampleFile}> ${$l("Download the Sample File")} </a>.
                 </div>
 
+                <pl-scroller
+                    class="stretch"
+                    ?hidden=${this._formatSelect && this._formatSelect.value !== imp.CSV.value}
+                >
+                    <ul class="vertical spacing layout">
+                        <div class="padded">
+                            <label>
+                                <input type="checkbox" id="csvHasDataOnFirstRowCheckbox" />
+                                ${$l("Check if the CSV file has data on the first row (instead of the column names)")}
+                            </label>
+                        </div>
+
+                        <pl-select
+                            id=${"nameColumnSelect"}
+                            .label=${$l("Name Column")}
+                            .options=${this._itemColumns.map((itemColumn, itemColumnIndex) => ({
+                                label: `${itemColumn.displayName} (${$l("Column {0}", itemColumnIndex.toString())})`,
+                                value: itemColumnIndex,
+                            }))}
+                            .selectedIndex=${this._nameColumnSelect?.selectedIndex}
+                            @change=${() => {
+                                const currentNameColumnIndex = this._itemColumns.findIndex(
+                                    (itemColumn) => itemColumn.type === "name"
+                                );
+                                const nameColumnIndex = this._nameColumnSelect?.value || 0;
+                                this._itemColumns[nameColumnIndex].type = "name";
+
+                                if (currentNameColumnIndex !== -1) {
+                                    this._itemColumns[currentNameColumnIndex].type = FieldType.Text;
+                                }
+
+                                this._parseData();
+                            }}
```

Comment:
For anything longer than a one-liner, let's use methods instead of inline functions.

Reply:
Thanks for the reminder, I'll still clean this all up before I'm done.